apps/src/templates/VersionRow.jsx

```diff
@@ -21,58 +26,101 @@ export default class VersionRow extends React.Component {
     return timestamp.toString();
   }
 
+  getUrlAttributes() {
+    const userId = queryParams('user_id');
+    const viewAs = queryParams('viewAs');
+    const sectionId = queryParams('section_id');
+    const params = {};
+    if (sectionId) {
+      params.section_id = sectionId;
+    }
+    if (viewAs) {
+      params.viewAs = viewAs;
+    }
+    if (userId) {
+      params.user_id = userId;
+    }
+    if (!this.props.isLatest) {
+      params.version = this.props.versionId;
+    }
+    return queryString.stringify(params);
+  }
+
   render() {
-    let button;
+    let buttons = [];
     if (this.props.isLatest) {
-      button = (
+      buttons.push(
         <button
+          key={'latest-version-message'}
           type="button"
           className="btn-default"
           disabled="disabled"
-          style={{cursor: 'default'}}
+          style={{cursor: 'default', background: 'none', border: 'none'}}
         >
-          {msg.currentVersion()}
+          {msg.latestVersion()}
         </button>
       );
-    } else {
-      button = [
+    } else if (!this.props.isReadOnly) {
+      const className = this.props.isSelectedVersion
+        ? 'btn-info'
+        : 'img-upload';
+      buttons.push(
+        <button
+          key={'restore-version-button'}
+          type="button"
+          className={className}
+          onClick={this.props.onChoose}
+        >
+          {msg.restore()}
+        </button>
+      );
+    }
+
+    if (!this.props.isSelectedVersion) {
+      buttons.push(
         <a
-          key={0}
+          key={'not-selected-version-button'}
           href={
-            location.origin +
-            location.pathname +
-            '?version=' +
-            this.props.versionId
+            location.origin + location.pathname + '?' + this.getUrlAttributes()
```

Comment:
Is it possible for `getUrlAttributes` to return an empty string?  If so, consider moving the "?" into the getUrlAttributes() so there's no extra "?" when there are no query params.

Reply:
It should never return an empty string (always either a teacher viewing student work so there will be several params, or a student viewing with their userid and/or version)